Merge microhttpd to develop

Regression is required before merging.

3 tests failed and 3 tests error for regression test:

```
======================================================================                                                                                [21/1813]
ERROR: test_find_transactions_by_tag (__main__.Regression_Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/regression/runner.py", line 371, in test_find_transactions_by_tag
    res_json["hashes"][0])
KeyError: 'hashes'

======================================================================
ERROR: test_find_transactions_obj_by_tag (__main__.Regression_Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/regression/runner.py", line 507, in test_find_transactions_obj_by_tag
    res_json.append(res_tx_json["transactions"][j])
KeyError: 'transactions'

======================================================================
ERROR: test_get_transactions_object (__main__.Regression_Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/regression/runner.py", line 433, in test_get_transactions_object
    self.assertEqual(sent_transaction_hash, res_json["hash"])
KeyError: 'hash'

======================================================================
FAIL: test_mam_recv_msg (__main__.Regression_Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/regression/runner.py", line 201, in test_mam_recv_msg
    self.assertTrue(expect_cases[i] in response[i]["content"])
AssertionError: False is not true

======================================================================
FAIL: test_mam_send_msg (__main__.Regression_Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/regression/runner.py", line 162, in test_mam_send_msg
    self.assertEqual(STATUS_CODE_500, response[i]["status_code"])
AssertionError: '500' != '404'
- 500
    self.assertEqual(sent_transaction_hash, res_json["hash"])
KeyError: 'hash'

======================================================================
FAIL: test_mam_recv_msg (__main__.Regression_Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/regression/runner.py", line 201, in test_mam_recv_msg
    self.assertTrue(expect_cases[i] in response[i]["content"])
AssertionError: False is not true

======================================================================
FAIL: test_mam_send_msg (__main__.Regression_Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/regression/runner.py", line 162, in test_mam_send_msg
    self.assertEqual(STATUS_CODE_500, response[i]["status_code"])
AssertionError: '500' != '404'
- 500
+ 404

======================================================================
FAIL: test_send_trytes (__main__.Regression_Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/regression/runner.py", line 746, in test_send_trytes
    self.assertEqual(STATUS_CODE_500, response[i]["status_code"])
AssertionError: '500' != '400'
- 500
? ^
+ 400
? ^


----------------------------------------------------------------------
Ran 10 tests in 41.341s

FAILED (failures=3, errors=3)
```

--- a/accelerator/http.c
+++ b/accelerator/http.c
@@ -73,12 +73,14 @@ static int set_response_content(status_t ret, char **json_result) {
   }
 
   cJSON *json_obj = cJSON_CreateObject();
-  switch (ret & SC_ERROR_MASK) {
-    case 0x03:
+  switch (ret) {
+    case SC_CCLIENT_NOT_FOUND:
+    case SC_MAM_NOT_FOUND:
       http_ret = MHD_HTTP_NOT_FOUND;
       cJSON_AddStringToObject(json_obj, "message", "Request not found");
       break;
-    case 0x07:
+    case SC_CCLIENT_JSON_KEY:
+    case SC_MAM_NO_PAYLOAD:
       http_ret = MHD_HTTP_BAD_REQUEST;
       cJSON_AddStringToObject(json_obj, "message", "Invalid request header");
       break;

